pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TenantInfo.java

```diff
@@ -19,13 +19,20 @@
 package org.apache.pulsar.common.policies.data;
 
 import java.util.Set;
+import org.apache.pulsar.client.admin.utils.ReflectionUtils;
 
 public interface TenantInfo {
     Set<String> getAdminRoles();
 
-    void setAdminRoles(Set<String> adminRoles);
-
     Set<String> getAllowedClusters();
 
-    void setAllowedClusters(Set<String> allowedClusters);
+    interface Builder {
+        Builder adminRoles(Set<String> adminRoles);
+        Builder allowedClusters(Set<String> allowedClusters);
+        TenantInfo build();
+    }
+
+    static Builder builder() {
+        return ReflectionUtils.newBuilder("org.apache.pulsar.common.policies.data.TenantInfoImpl");
```

Comment:
Whether should we put `rg.apache.pulsar.common.policies.data.TenantInfoImpl` into `pulsar-client-admin-api` module instead of `pulsar-common`?

Reply:
The reason for it is the presence of Swagger annotations which cannot be put as dependency of the `pulsar-client-admin-api` module.

Reply:
We should make sure to use `pulsar-client-admin-api` dependent project must include `pulsar-common` dependency. Otherwise, it will throw `no such class exception`

Reply:
That's correct. You won't use `pulsar-client-admin-api` directly, but through `pulsar-client-admin` which is a shaded package, or in the context of functions, the `-api` module will be exposed to the function while the `pulsar-client-admin` will be kept in the framework class loader.